PubSub/src/BatchPublisher.php

```diff
@@ -82,27 +108,86 @@ public function __construct($topicName, array $options = [])
      * ]);
      * ```
      *
-     * @param array $message [Message Format](https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage).
+     * @param Message|array $message An instance of
+     *        {@see Google\Cloud\PubSub\Message}, or an array in the correct
+     *        [Message Format](https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage).
      * @return bool
      */
-    public function publish(array $message)
+    public function publish($message)
     {
-        return $this->batchRunner->submitItem($this->identifier, $message);
+        $message = $message instanceof Message
+            ? $message->toArray()
+            : $message;
+
+        $job = $this->getJob($message);
+
+        return $job->publish($message);
     }
 
     /**
      * Returns an array representation of a callback which will be used to write
      * batch items.
      *
      * @return array
+     * @deprecated
      */
     protected function getCallback()
+    {
+        return $this->batchMethod;
+    }
+
+    /**
+     * Intended for internal use only by the batch publisher.
+     *
+     * This method is called internally by the batch runner.
+     * {@see Google\Cloud\PubSub\OrderingKeyBatchJob} provides this method as
+     * its publish callback.
+     *
+     * @see https://cloud.google.com/pubsub/docs/reference/rest/v1/projects.topics/publish Publish Message
+     *
+     * @param array[] $messages A list of messages. Each message must be in the correct
+     *        [Message Format](https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage).
+     * @param array $options [optional] Configuration Options
+     * @return array A list of message IDs.
+     * @internal
+     * @access private
+     */
+    public function publishDeferred(array $messages, array $options = [])
     {
         if (!array_key_exists($this->topicName, self::$topics)) {
-            $client = new PubSubClient($this->getUnwrappedClientConfig());
```

Comment:
I just wanted to confirm this is safe to remove, if we are serializing the config wouldn't this still be required?